source/src_pw/stress_func_ewa.cpp

```diff
@@ -99,59 +113,106 @@ void Stress_Func::stress_ewa(ModuleBase::matrix& sigma, ModulePW::PW_Basis* rho_
     double fac;
 	if(ig0 >= 0)
 	{
-		rmax = 4.0/sqrt(alpha)/GlobalC::ucell.lat0;
-		//with this choice terms up to ZiZj*erfc(5) are counted (erfc(5)=2*10^-1)
-		for(int it=0; it < GlobalC::ucell.ntype; it++)
+		r = new ModuleBase::Vector3<double>[mxr];
+		r2 = new double[mxr];
+		irr = new int[mxr];
+
+		double sqa = sqrt(alpha);
+		double sq8a_2pi = sqrt(8 * alpha / (ModuleBase::TWO_PI));
+		rmax = 4.0/sqa/GlobalC::ucell.lat0;
+
+		// collapse it, ia, jt, ja loop into a single loop
+		long long ijat, ijat_end;
+		ModuleBase::TASK_DIST_1D(num_threads, thread_id, (long long)GlobalC::ucell.nat * GlobalC::ucell.nat, ijat, ijat_end);
+		ijat_end = ijat + ijat_end;
+
+		long long iat = ijat / GlobalC::ucell.nat;
+		long long jat = ijat % GlobalC::ucell.nat;
+
+		int it = (ijat < ijat_end) ? GlobalC::ucell.iat2it[iat] : GlobalC::ucell.ntype;
+		int i = (ijat < ijat_end) ? GlobalC::ucell.iat2ia[iat] : 0;
+
+		int jt = (ijat < ijat_end) ? GlobalC::ucell.iat2it[jat] : GlobalC::ucell.ntype;
+		int j = (ijat < ijat_end) ? GlobalC::ucell.iat2ia[jat] : 0;
```

Comment:
I think loop for （it，ia）and loop for（it, ia, jt, ja）is two common case in ABACUS. 

So maybe the better solution is create interfaces function from "GlobalC::ucell" to get all indexes for one thread, but not modify them one by one with repete code.

Reply:
I think your right, and there is a easy way to do this.
I will open another PR for it